File: json_to_flutter/functions/jsonToWidgetList.dart
import 'dart:convert';
import 'dart:io';
import '../models/index.dart';
import '../debug.dart';

Future<List<Widget>> jsonToWidgetList() async {
  List<Widget> widgets = [];
  try {
    await new File('widgets.json').readAsString().then((fileContents) async {
      var parsedJson = json.decode(fileContents);
      var aux;
      for (int i = 0; i < 9999999; i++) {
        aux = _addWidgetOnList(parsedJson["$i"]);
        if (aux == null) break;
        widgets.add(aux);
      }
    });
  } catch (e) {
    print(red("Without JSON"));
    return null;
  }
  return widgets;
}

Widget _addWidgetOnList(var json) {
  if (json == null) return null;
  switch (json["type"]) {
    case "container":
      return Container.fromJson(json);
    case "svg":
      return Svg.fromJson(json);
    default:
      return Text.fromJson(json);
  }
}

File: src/functions/widget/jsonToWidgetList.js
const { Container } = require("./models/container");
const { Svg } = require("./models/svg");
const { Text } = require("./models/text");

async function jsonToWidgetList(json) {
  let widgets = [];
  try {
    json = JSON.parse(json);
    for (let i = 0; i < 9999999; i++) {
      let widget = _addWidgetOnList(json[`${i}`]);
      if (widget == null) break;
      widgets.push(widget);
    }
  } catch (e) {
    console.log("Without JSON");
    return null;
  }
  return widgets;
}

function _addWidgetOnList(json) {
  if (json == null) return null;
  switch (json["type"]) {
    case "container":
      return new Container(json);
    case "svg":
      return new Svg(json);
    default:
      return new Text(json);
  }
}

module.exports = { jsonToWidgetList };